lib/schema/utils.js

```diff
@@ -153,10 +156,370 @@ const parseNestedPropertyForResourceAndField = ({ arr, isValueArray }) => {
+
+  return f?.split('@')[0];
+};
+
+const getValidDataDictionaryVersions = () => Object.values(DATA_DICTIONARY_VERSIONS ?? {});
+const isValidDataDictionaryVersion = (version = '') => getValidDataDictionaryVersions()?.includes(version);
+
+/**
+ * Combines results from one or more validations into one object.
+ *
+ * @param {Object} obj
+ * @param {{}} obj.errorCache
+ * @param {{}} obj.warningsCache
+ * @param {{}} obj.payloadErrors
+ * @param {{totalWarnings: number, totalErrors: number}} obj.stats
+ * @returns {Object} Consolidated error and warning report
+ */
+const combineErrors = ({ errorCache, warningsCache, payloadErrors, stats }) => {
+  const errorReport = [];
+
+  // Helper function to process either errors or warnings from cache
+  const processCache = (cache, type) => {
+    Object.entries(cache).forEach(([resource, fields]) => {
+      Object.entries(fields).forEach(([fieldName, errors]) => {
+        Object.entries(errors).forEach(([, details]) => {
+          if (!details) return;
+
+          const occurrences = Object.entries(details).map(([fileKey, fileDetails]) => ({
+            count: fileDetails.occurrences,
+            ...parseFileName(fileKey),
+            message: details?.[fileKey]?.message
+          }));
+          const message = occurrences[0]?.message || '';
+          addReport(
+            errorReport,
+            resource,
+            fieldName,
+            type,
+            message,
+            occurrences.map(({ message: _, ...rest }) => rest)
+          );
+        });
+      });
+    });
+  };
+
+  // Helper function to parse the filename for any given key pattern like '<file>__<lookup>'
+  const parseFileName = fileKeyName => {
+    const [fileName, lookupValue] = fileKeyName.split('__');
+    return {
+      ...(fileName && { fileName }),
+      ...(lookupValue && { lookupValue })
+    };
+  };
+
+  // Helper function to add error/warning data to errorReport
+  const addReport = (report, resource, fieldName, type, message, occurrences) => {
+    let fieldData = report.find(x => x.resourceName === resource && x.fieldName === fieldName);
+    if (!fieldData) {
+      fieldData = { resourceName: resource, fieldName: fieldName, errors: [], warnings: [] };
+      report.push(fieldData);
+    }
+    fieldData[type].push({ message, occurrences });
+  };
+
+  // Process errors and warnings
+  processCache(errorCache, 'errors');
+  processCache(warningsCache, 'warnings');
+
+  // Process payload errors
+  Object.entries(payloadErrors).forEach(([resource, messages]) => {
+    Object.entries(messages).forEach(([message, files]) => {
+      const occurrences = files.map(fileName => ({
+        count: 1, // Static count since payload errors are listed individually
+        fileName
+      }));
+      addReport(errorReport, resource, '', 'errors', message, occurrences);
+      stats.totalErrors++;
+    });
+  });
+
+  return {
+    description: 'RESO Common Format Schema Validation Summary',
```

Comment:
Add `version` to the report so we know which DD version it was validated against.